control-plane/pkg/reconciler/broker/broker_config.go

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2020 The Knative Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package broker
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/Shopify/sarama"
+	"go.uber.org/zap"
+	corev1 "k8s.io/api/core/v1"
+	"knative.dev/pkg/configmap"
+)
+
+type Config struct {
+	TopicDetail      sarama.TopicDetail
+	BootstrapServers []string
+}
+
+func configFromConfigMap(logger *zap.Logger, cm *corev1.ConfigMap) (Config, error) {
+
+	topicDetail := sarama.TopicDetail{}
+
+	var replicationFactor int32
+	var bootstrapServers string
+
+	err := configmap.Parse(cm.Data,
+		configmap.AsInt32(DefaultTopicNumPartitionConfigMapKey, &topicDetail.NumPartitions),
+		configmap.AsInt32(DefaultTopicReplicationFactorConfigMapKey, &replicationFactor),
+		configmap.AsString(BootstrapServersConfigMapKey, &bootstrapServers),
+	)
+	if err != nil {
+		return Config{}, fmt.Errorf("failed to parse config map %s/%s: %w", cm.Namespace, cm.Name, err)
+	}
+
+	if topicDetail.NumPartitions <= 0 || replicationFactor <= 0 || bootstrapServers == "" {
```

Comment:
Are you sure these validations can't be done directly from `configmap.Parse`?

Reply:
I don't think so, but I'm thinking to write a `ParseFunc` for each config field, which actually encapsulates the validation, so that while we add new configurations this if doesn't become unreadable, WDYT?

Reply:
I'm fine with that, but let's do it later, because it requires changes to pkg